src/network.c

```diff
@@ -194,36 +193,44 @@ static int network_server_init(server *srv, buffer *host_token, specific_config
 	b = buffer_init();
 	buffer_copy_buffer(b, host_token);
 
-	/* ipv4:port
-	 * [ipv6]:port
-	 */
-	if (NULL == (sp = strrchr(b->ptr, ':'))) {
-		log_error_write(srv, __FILE__, __LINE__, "sb", "value of $SERVER[\"socket\"] has to be \"ip:port\".", b);
+	host = b->ptr;
 
-		goto error_free_socket;
-	}
+	if (host[0] == '/') {
+		/* host is a unix-domain-socket */
+		is_unix_domain_socket = 1;
+	} else {
+		/* ipv4:port
+		 * [ipv6]:port
+		 */
+		size_t len = buffer_string_length(b);
+		char *sp = NULL;
+		if ((b->ptr[0] == '[' && b->ptr[len-1] == ']') || NULL == (sp = strrchr(b->ptr, ':'))) {
+			if (NULL == sp) {
```

Comment:
I think this condition (`NULL == sp`) is always true; if `(b->ptr[0] == '[' && b->ptr[len-1] == ']')` the second part of `||` doesn't get evaluated and `sp` is still `NULL`; otherwise the second part was true and `NULL == (sp = ...)`.